urts/src/lib.rs

```diff
@@ -93,35 +179,77 @@ impl EnclaveBuilder {
                 self.debug as c_int,
                 &mut enclave_id,
                 ptr::null_mut(),
-                0,
-                ptr::null_mut(),
+                ex_features,
+                &mut ex_features_p as *mut *const c_void,
             )
-        };
-        match result {
-            sgx_status_t::SGX_SUCCESS => Ok(Enclave::new(enclave_id)),
-            error => Err(Error::SgxStatus(error)),
         }
+        .into_result()
+        .map_err(Error::from)
+        .map(|_| Enclave { id: enclave_id })
     }
 }
 
 impl From<Vec<u8>> for EnclaveBuilder {
-    fn from(bytes: Vec<u8>) -> EnclaveBuilder {
+    fn from(input: Vec<u8>) -> EnclaveBuilder {
         EnclaveBuilder {
-            bytes,
+            bytes: input,
             debug: false,
+            pcl_key: None,
+            kss_config: None,
         }
     }
 }
 
-impl Enclave {
-    fn new(id: sgx_enclave_id_t) -> Enclave {
-        Enclave { id }
+impl From<&[u8]> for EnclaveBuilder {
+    fn from(input: &[u8]) -> EnclaveBuilder {
+        EnclaveBuilder {
+            bytes: input.to_vec(),
+            debug: false,
+            pcl_key: None,
+            kss_config: None,
+        }
+    }
+}
+
+impl<const N: usize> From<&[u8; N]> for EnclaveBuilder {
+    fn from(input: &[u8; N]) -> EnclaveBuilder {
+        EnclaveBuilder {
+            bytes: input.to_vec(),
+            debug: false,
+            pcl_key: None,
+            kss_config: None,
+        }
     }
 }
 
-impl Deref for Enclave {
-    type Target = sgx_enclave_id_t;
-    fn deref(&self) -> &Self::Target {
+impl TryFrom<File> for EnclaveBuilder {
```

Comment:
I think `impl<P: AsRef<Path>> TryFrom<&'a P> for EnclaveBuilder` may be the way to do this.

Reply:
I can drop the `TryFrom` for `File`. It's in there because I hate being in a situation where I have a `Path` or a `File` but the API I'm talking to requires me to only use the other one. One concrete example of this is when testing using temporary files, it's often much more annoying (for valid security-type reasons) to get a `NamedTempFile` than to get a `TempFile`.

Reply:
Per discussion with James, keeping `TryFrom<File>`, and as said above, can't do `TryFrom` with generics so leaving it in `new`.